proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,83 @@
+
+This is further complicated when developers mix local and machine wide installations. The host resolver will find the first `dotnet` according to its lookup rules and search only there for a compatible SDK. Once developers manipulate `%PATH%` to prefer local SDKS the resolver will stop considering machine wide SDKS. That can lead to situations where there is machine wide SDK that works for a given global.json but the host resolver will not consider it because the developer setup `%PATH%` to consider a locally installed SDK. That can be very frustrating for end users.
+
+This disconnect between the resolver and deployment has lead to customers introducing a number of creative work arounds:
+
+- [scripts](https://github.com/dotnet/razor/pull/9550) to launch VS Code while considering locally deployed .NET SDKs
+- [docs and scripts](https://github.com/dotnet/sdk/blob/518c60dbe98b51193b3a9ad9fc44e055e6e10fa0/documentation/project-docs/developer-guide.md?plain=1#L38) to setup the environment and launch VS so it can find the deployed .NET SDKs.
+- [scritps](https://github.com/dotnet/runtime/blob/main/dotnet.cmd) that wrap `dotnet` to find the  _correct_ `dotnet` to use during build.
+
+These scripts are not one offs, they are increasingly common items in repos in `github.com/dotnet` to attempt to fix the disconnect. Even so many of these solutions are incomplete because they themselves only consider local deployment. They dont't fully support the full set of ways the SDK can be deployed.
+
+This is not a problem specific to the .NET team. Indeed this problem is felt sharply there due to arcade infrastructure using xcopy style deployment into `.dotnet`. Our customers feel this problem as well. Consider the following examples:
+
+- This [issue](https://github.com/dotnet/sdk/issues/8254) from 2017 attempting to solve this problsem. It gets several hits a year from customers who are similarly struggling with our toolings inability to handle local deployment.
+- This [internal discussion](https://teams.microsoft.com/l/message/19:ed7a508bf00c4b088a7760359f0d0308@thread.skype/1698341652961?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4ba7372f-2799-4677-89f0-7a1aaea3706c&parentMessageId=1698341652961&teamName=.NET%20Developer%20Experience&channelName=InfraSwat&createdTime=1698341652961) from a C# team member. They wanted to use VS as the product is shipped to customers and got blocked when we shipped an SDK that didn't have a corresponding MSI and hence VS couldn't load Roslyn anymore.
+- [VS code](https://github.com/dotnet/vscode-csharp/issues/6471) having to adjust to consider local directories for SDK because our resolver can't find them.
```

Comment:
These two are internal .NET teams, are they not? "Customer impact" is still the sticking point for me on this.

Reply:
I was viewing it more as people just trying to use the product and running into local deployment issues. It's not strictly related to their positions on the .NET team. Can be frustrating when you're trying to use the prooduct as a customer but get labeled as "it's part of .NET so think about it differently". Was trying to thread that needle in my wording here. 

I can rephrase this as customre impact though. It's a better term.

Reply:
My position remains that if you're using a global.json with no rollforward you're not just "trying to use the product". Someone opted you into specific pain to lock down behavior. Is that justified? Maybe! But it's not required for everyone.

Reply:
> My position remains that if you're using a global.json with no rollforward you're not just "trying to use the product".

I don't agree with this point. There are plenty of customers out there who desire a very locked down build environment. As compatible as the .NET SDK is, upgrading can cause problems, particularly with complex projects. Also for people who are eagerly dogfooding preview versions of the SDK, disabling roll forward seems helpful if they're playing with different previews in different projects. These all seem like valid customer scenarios. 

At the same time, a large part of the motivation for disabling `rollForward` would actually go away with this feature. Generally this is done for CI scenarios where we want to avoid machine state making CI unpredictable. Once we have `additionalPaths` that problem kinda goes away because we know that will be matched first. Machine state doesn't matter as much.

Reply:
> These all seem like valid customer scenarios.

I think they are, but I don't think the evidence you've presented is convincing that they're common.

> At the same time, a large part of the motivation for disabling `rollForward` would actually go away with this feature. Generally this is done for CI scenarios where we want to avoid machine state making CI unpredictable.

This is surprising to me, since I thought it was the inverse: CI is fine and VS is where this feature would benefit. Why not solve this with "use whatever magic you used to create the `.dotnet` folder to execute the build"?

Reply:
Okay tried to clean that up a bit.